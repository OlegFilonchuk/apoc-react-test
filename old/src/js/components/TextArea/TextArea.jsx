import React from 'react';
import PropTypes from 'prop-types';

import './TextArea.less';

function TextArea({ className, ...props }) {
  return <textarea {...props} className={`text-area sc-form-ele ${className}`} />;
}

TextArea.propTypes = {
  /**
   * Additional class names
   */
  className: PropTypes.string
};

TextArea.defaultProps = {
  className: ''
};

export default TextArea;
