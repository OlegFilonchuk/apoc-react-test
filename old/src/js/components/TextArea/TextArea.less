@import '../../utils/mixins/formElements';

.text-area {
  resize: none;

  .form-element;
}
