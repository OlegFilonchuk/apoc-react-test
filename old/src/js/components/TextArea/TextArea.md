## TextArea examples

Simple usage of the **TextArea** component:

    (
      <div>
        <TextArea />
      </div>
    )

**TextArea** with default value (uncontrolled):

    (
      <div>
        <TextArea defaultValue="some default value"/>
      </div>
    )

**TextArea** with placeholder:

    (
      <div>
        <TextArea placeholder="Enter your description here"/>
      </div>
    )

**TextArea** as controlled component:

    const initialState = {
      value: 'TextArea text here'
    };

    const onChange = e => {
      setState({ value: e.target.value })
    };

    (
      <div>
        <TextArea
          onChange={onChange}
          value={state.value}
        />
      </div>
    )
