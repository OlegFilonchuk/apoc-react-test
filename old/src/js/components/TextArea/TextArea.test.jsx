import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextArea from './TextArea';

describe('<TextArea>', () => {
  const mountTextArea = (props, mountFn = mount) => mountFn(<TextArea {...props} />);

  it('should render TextArea component', () => {
    const props = { defaultValue: 'test' };

    const component = mountTextArea(props, shallow);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should add placeholder to TextArea', () => {
    const props = { defaultValue: 'test', placeholder: 'placeholder' };

    const component = mountTextArea(props, shallow);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should run onBlur callback when TextArea is changed', () => {
    const onBlur = jest.fn();

    const props = {
      onBlur,
      defaultValue: 'test'
    };

    const expectedValue = 'test2';

    const component = mountTextArea(props);

    component.find('textarea').simulate('blur', {
      target: {
        value: expectedValue
      }
    });

    expect(onBlur).toHaveBeenCalledWith(
      expect.objectContaining({
        target: {
          value: expectedValue
        }
      })
    );
  });

  it('should run onChange callback when TextArea is changed', () => {
    const onChange = jest.fn();

    const props = {
      onChange,
      defaultValue: 'test'
    };
    const expectedValue = 'test2';

    const component = mountTextArea(props);

    component.find('textarea').simulate('change', {
      target: {
        value: expectedValue
      }
    });

    expect(onChange).toHaveBeenCalledWith(
      expect.objectContaining({
        target: {
          value: expectedValue
        }
      })
    );
  });

  it('should update content when receive props', () => {
    const onBlur = jest.fn();

    const props = {
      onBlur,
      defaultValue: 'test'
    };

    const expectedValue = 'test2';
    const component = mountTextArea(props);

    expect(component.find('textarea').text()).toEqual('test');

    component.setProps({ defaultValue: expectedValue });

    expect(component.find('textarea').text()).toEqual(expectedValue);
  });
});
