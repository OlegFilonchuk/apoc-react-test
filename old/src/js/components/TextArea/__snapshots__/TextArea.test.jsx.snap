// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea> should add placeholder to TextArea 1`] = `
<textarea
  className="text-area sc-form-ele "
  defaultValue="test"
  placeholder="placeholder"
/>
`;

exports[`<TextArea> should render TextArea component 1`] = `
<textarea
  className="text-area sc-form-ele "
  defaultValue="test"
/>
`;
